File: backend/src/main/java/com/naviksha/model/CareerMatch.java
package com.naviksha.model;

import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CareerMatch {
    private String careerName;
    private Integer matchScore;
    private List<String> topReasons;
    private List<String> studyPath;
    private List<String> first3Steps;
    private String confidence;
    private String whatWouldChangeRecommendation;
}

File: ai-report-service/app/models/career_match.py
from pydantic import BaseModel
from typing import List, Optional

class CareerMatch(BaseModel):
    """Career match data structure"""
    career_name: str
    match_score: int
    bucket: str
    riasec_profile: str
    primary_subjects: List[str]
    top_reasons: List[str]
    study_path: List[str]
    first3_steps: List[str]
    confidence: str
    what_would_change_recommendation: str
